src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -114,6 +114,11 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
+        if ($this->registered && null !== $this->dispatcher) {
```

Comment:
I'm not sure if this was right choice. Technically the ContextListener might be registered without a dispatcher as it an optional constructor argument. We do check for its existence in the `handle()` method.